### PR TITLE
Added a new flag `sendIfEmpty` for pings

### PR DESCRIPTION
**Pull request by badboy**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry to `HISTORY.rst` or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: glean_parser/schemas/pings.1-0-0.schema.yaml [hunk 1]
```diff
@@ -54,6 +54,13 @@ additionalProperties:
         When `true`, include the `client_id` value in the ping.
       type: boolean
 
+    send_if_empty:
+      title: Send if empty
+      description: |
+        When `false` (the default) a ping is send only if it contains data.
```
Comment (Dexterp37) [inline glean_parser/schemas/pings.1-0-0.schema.yaml]:
```suggestion
        When `false` (the default) a ping is sent only if it contains data.
```

Comment (Dexterp37) [reply glean_parser/schemas/pings.1-0-0.schema.yaml]: Note: I might be wrong, but it sounded weird. Maybe a @chutten check?

File: glean_parser/schemas/pings.1-0-0.schema.yaml [hunk 2]
```diff
@@ -54,6 +54,13 @@ additionalProperties:
...
+        When `true` a ping is send even if it contains no data.
```
Comment (Dexterp37) [inline glean_parser/schemas/pings.1-0-0.schema.yaml]:
```suggestion
        When `true` a ping is sent even if it contains no data.
```

File: glean_parser/schemas/pings.1-0-0.schema.yaml (hunk 1 above)
Comment (travis79) [reply glean_parser/schemas/pings.1-0-0.schema.yaml]: "a ping is sent" is correct if you are asking about English grammar

**Pull request merged**